File: 214/projects/10/java/Birds.java
/* Birds.java illustrates inheritance and polymorphism.
 *
 * Begun by: Joel Adams, for CS 214 Lab 10 at Calvin College.
 * Completed by: Duncan Van Keulen
 * Date: 4/23/2020
 *********************************************************/

public class Birds {
  
  public static void main(String[] args) {
    System.out.println("\nWelcome to the Bird Park!\n");

    WalkingBird bird01 = new WalkingBird();
    FlyingBird bird02 = new FlyingBird();

    bird01.print();
    bird02.print();

    Duck bird2 = new Duck("Donald");
    bird2.print();

    Goose bird3 = new Goose("Mother Goose");
    bird3.print();

    Owl bird4 = new Owl("Woodsey");
    bird4.print();

    Penguin bird5 = new Penguin("Peter");
    bird5.print();
    
    Ostrich bird6 = new Ostrich("Orville");
    bird6.print();
    
    Kiwi bird7 = new Kiwi("Kathy");
    bird7.print();
    System.out.println();

  }
}

File: code/projects/10/ruby/birds.rb
# birds.rb | Tests out the menagerie of bird classes.
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Duncan Van Keulen
# Date: 4/27/2020
######################################################

require './Bird.rb'
require './Duck.rb'
require './Goose.rb'
require './Owl.rb'
require './FlyingBird.rb'
require './WalkingBird.rb'
require './Penguin.rb'
require './Ostrich.rb'
require './Kiwi.rb'

puts "\nWelcome to the Bird Park!\n\n"

bird1 = Duck.new "Donald"
bird1.print

bird2 = Goose.new "Mother"
bird2.print

bird3 = Owl.new "Woodsey"
bird3.print

bird4 = FlyingBird.new "Average Flyer"
bird4.print

bird5 = WalkingBird.new "Average Walker"
bird5.print

bird6 = Penguin.new "Peter"
bird6.print

bird7 = Ostrich.new "Orville"
bird7.print

bird8 = Kiwi.new "Kathy"
bird8.print

puts "\n\n"